ruby-head-wasm-wasi build fails on current Ruby head because cross build runs wasm dump_ast instead of host dump_ast

## Summary

Building `ruby-head-wasm-wasi` against current Ruby `head` fails during `.rbinc` generation in the cross build.
The failure occurs because the build executes `./dump_ast`, which is a wasm binary, instead of a host-built `dump_ast`.

Although Ruby `head` supports providing a host `dump_ast` via `--with-dump-ast`, the generated `Makefile` still uses the target-side binary when it is not specified.

## Environment

The Ruby source used by the failing build was:

- https://github.com/ruby/ruby/commit/251fd7a11e442c34c1b60908409e5611f69ef858
 
This reproduces on a clean build with:

```sh
rm -rf build
rake npm:ruby-head-wasm-wasi
```

Results:

```text
./dump_ast: 151: Syntax error: "(" unexpected
make: *** [uncommon.mk:1307: /home/ledsun/ruby.wasm/build/checkouts/head/io.rbinc] Error 1
Try running with `rake --verbose` for more complete output.
bundler: failed to load command: /home/ledsun/ruby.wasm/exe/rbwasm (/home/ledsun/ruby.wasm/exe/rbwasm)
/home/ledsun/ruby.wasm/lib/ruby_wasm/build/executor.rb:77:in 'RubyWasm::BuildExecutor#system': Command failed with status (2): 'make' '-j8' 'install' 'DESTDIR=/home/ledsun/ruby.wasm/build/wasm32-unknown-wasip1/ruby-head-wasm32-unknown-wasip1-full-61ff6b0c74de0393d023cecfb9d1d7be/install' (RuntimeError)
        from /home/ledsun/ruby.wasm/lib/ruby_wasm/build/product/crossruby.rb:238:in 'RubyWasm::CrossRubyProduct#build'
```

## Related change

This behavior appears after:

- https://github.com/ruby/ruby/commit/a70dbe1065f58a23a85e576d19e3434e6dd4bb64

This change introduces support for `--with-dump-ast`, but the fallback behavior still uses `./dump_ast` even in cross builds.
The generated `Makefile` contains:

```make
DUMP_AST = ./dump_ast$(EXEEXT)
```
As a result, `.rbinc` generation invokes the target-side `dump_ast` instead of a host executable.

## Proposed fix

Pass the host-built `dump_ast` explicitly to configure:

`--with-dump-ast=/path/to/host/baseruby-build/dump_ast`

This ensures `.rbinc` generation uses a host executable.

`ruby.wasm` already passes `--with-baseruby`, so it can also pass `--with-dump-ast`.

Minimal patch:

```diff
diff --git a/lib/ruby_wasm/build/product/crossruby.rb b/lib/ruby_wasm/build/product/crossruby.rb
@@
     def baseruby_path
       File.join(@baseruby.install_dir, "bin/ruby")
     end
+
+    def dump_ast_path
+      File.join(@baseruby.product_build_dir, "dump_ast")
+    end
@@
       args << %Q(--with-zlib-dir=#{@zlib.install_root})
       args << %Q(--with-openssl-dir=#{@openssl.install_root}) if @openssl
       args << %Q(--with-baseruby=#{baseruby_path})
+      args << %Q(--with-dump-ast=#{dump_ast_path})
```

This resolves the issue by avoiding execution of the `wasm dump_ast`.

--- a/lib/ruby_wasm/build/product/crossruby.rb
+++ b/lib/ruby_wasm/build/product/crossruby.rb
@@ -322,6 +322,10 @@ module RubyWasm
       File.join(@baseruby.install_dir, "bin/ruby")
     end
 
+    def dump_ast_path
+      File.join(@baseruby.product_build_dir, "dump_ast")
+    end
+
     def configure_args(build_triple, toolchain)
       target = @params.target.triple
       default_exts = @params.default_exts
@@ -336,6 +340,9 @@ module RubyWasm
       args << %Q(--with-zlib-dir=#{@zlib.install_root})
       args << %Q(--with-openssl-dir=#{@openssl.install_root}) if @openssl
       args << %Q(--with-baseruby=#{baseruby_path})
+      # Use the host-built dump_ast so cross builds don't try to execute the
+      # target-side wasm dump_ast while generating .rbinc files.
+      args << %Q(--with-dump-ast=#{dump_ast_path})
 
       case target
       when /^wasm32-unknown-wasi/

--- a/sig/ruby_wasm/build.rbs
+++ b/sig/ruby_wasm/build.rbs
@@ -218,6 +218,7 @@ module RubyWasm
     def extinit_obj: -> String
     def extinit_c_erb: -> String
     def baseruby_path: -> String
+    def dump_ast_path: -> String
     def configure_args: (String build_triple, Toolchain toolchain) -> Array[String]
     def rbconfig_rb: -> String?
   end

